### PR TITLE
Make workflow step name editable

### PR DESCRIPTION
**Pull request by thomtrp**
- Use TextInput in header title
- add onTitleChange prop
- rename field name instead of label

To fix :
- padding right on title comes from current TextInput component. It needs to be refactored

https://github.com/user-attachments/assets/535cd6d3-866b-4a61-9c5d-cdbe7710396a

### CODE REVIEW

File: packages/twenty-front/src/modules/ui/field/input/components/TextInput.tsx [hunk 1]
```diff
@@ -37,19 +46,20 @@ export const TextInput = ({
   onClickOutside,
   onChange,
   copyButton = true,
+  shouldTrim = true,
```
Comment (martmull) [inline packages/twenty-front/src/modules/ui/field/input/components/TextInput.tsx:49]: I don't understand why we need this parameter. Why should we not trim sometimes?

Comment (Devessier) [reply packages/twenty-front/src/modules/ui/field/input/components/TextInput.tsx:49]: The `<TextInput />` component has always been used where we don't want spaces to be filled in the input, like for the Number field. Now that we use it for a regular text field, it was impossible to add a space character.

**Comment by thomtrp** [comment]
Good catch @martmull 💪 

Updated
Capture d’écran 2024-11-22 à 14 51 14

File: packages/twenty-front/src/modules/ui/field/input/components/TextInput.tsx (hunk 1 above)
Comment (martmull) [reply packages/twenty-front/src/modules/ui/field/input/components/TextInput.tsx:49]: Still don't get it. <TextInput is used in FieldName or Name Singular, which allows adding spaces

**Review by martmull — APPROVED** [review]

**Pull request merged**